ui-performance/index.md

```diff
@@ -0,0 +1,484 @@
+   This thread includes code that you wrote, and code executed by
+   Flutter's framework on your app's behalf.
+   When your app cretes and displays a scene, the UI thread creates
+   a _layer tree_, a lightweight object containing device-agnostic
+   painting commands, and sends the layer tree to the GPU thread to
+   be rendered on the device. _Don't block this thread!_
+
+1. GPU thread<br>
+   The GPU thread takes the layer tree and displays it by talking
+   to the GPU (graphic processing unit). You cannot directly access
+   the GPU thread or its data but, if this thread is slow,
+   it's a result of something you've done in the Dart code.
+   Skia, the graphics library, runs on this thread, which is sometimes
+   called the _rasterizer_ thread.
+
+1. I/O thread<br>
+   Performs expensive tasks (mostly I/O) that would otherwise block
+   either the UI or GPU threads.
+
+For more information, see
+[Architecture notes.](https://github.com/flutter/engine/wiki#architecture-notes)
+
+Each graph in the performance overlay represents the last 300 frames
+for the UI and GPU threads:
+
+* Rasterizer<br>
+  The rasterizer graph (on the top) reflects activity in the GPU thread.
+
+* Engine<br>
+  The engine graph (on the bottom) reflects activity in the UI thread.
+
+Each frame should be created and displayed within 16ms
+(or 1/60th of a second).  A frame exceeding this limit (in either graph)
+fails to display, resulting in jank, and a vertical red bar appears in one or
+both of the graphs.
+If a red bar appears in the engine graph, the Dart code is too expensive.
+If a red vertical bar appears in the rasterizer graph, the scene is too
+complicated to render quickly.
+
+<center><img src="images/performance-overlay-jank.png" al="Screenshot of performance overlay showing jank."></center>
+<center>The vertical red bar in the rasterier graph indicates that the current frame is expensive to paint.</center><br>
+
+### Displaying the performance overlay
+
+You can toggle display of the performance overlay as follows:
+
+* Using the Flutter Inspector
+* From the command line
+* Programmatically
+
+#### Flutter Inspector
+
+The easiest way to enable the PerformanceOverlay widget is by enabling it
+in the Flutter Inspector, which is available through the Flutter plugin
+for your IDE. The Inspector view opens by default when running an
+application. If the inspector is not open, you can display it as follows.
+
+In Android Studio and IntelliJ IDEA:
+
+1. Select **View > Tool Windows > Flutter Inspector**.
+1. In the toolbar, select the icon that looks like a bookshelf (<img src="images/performance-overlay-icon.png" alt="icon that resembles a bookshelf">).
+
+![IntelliJ Flutter Inspector Window](/images/intellij/visual-debugging.png)<br>
+
+The Flutter Inspector is available in the VS Code, Android Studio,
+and IntelliJ plugins, but its UI may change as it continues to evolve.
+The inspector is not currently available from the command line.
```

Comment:
There is some CLI support for the inspector. Whether we want to explicitly mention it or not is another question. /cc @jacob314